src/main/drivers/rangefinder/rangefinder_vl53l1x.c

```diff
@@ -0,0 +1,1693 @@
+    status = VL53L1X_StartRanging(dev);
+    while(tmp==0){
+        status = VL53L1X_CheckForDataReady(dev, &tmp);
+    }
+    tmp  = 0;
+    status = VL53L1X_ClearInterrupt(dev);
+    status = VL53L1X_StopRanging(dev);
+    status = VL53L1_WrByte(dev, VL53L1_VHV_CONFIG__TIMEOUT_MACROP_LOOP_BOUND, 0x09); /* two bounds VHV */
+    status = VL53L1_WrByte(dev, 0x0B, 0); /* start VHV from the previous temperature */
+    return status;
+}
+
+
+int8_t VL53L1X_CalibrateOffset(busDevice_t * dev, uint16_t TargetDistInMm, int16_t *offset)
+{
+    uint8_t i, tmp;
+    int16_t AverageDistance = 0;
+    uint16_t distance;
+    VL53L1X_ERROR status = 0;
+
+    status = VL53L1_WrWord(dev, ALGO__PART_TO_PART_RANGE_OFFSET_MM, 0x0);
+    status = VL53L1_WrWord(dev, MM_CONFIG__INNER_OFFSET_MM, 0x0);
+    status = VL53L1_WrWord(dev, MM_CONFIG__OUTER_OFFSET_MM, 0x0);
+    status = VL53L1X_StartRanging(dev); /* Enable VL53L1X sensor */
+    for (i = 0; i < 50; i++) {
+        tmp = 0;
+        while (tmp == 0){
+            status = VL53L1X_CheckForDataReady(dev, &tmp);
+        }
+        status = VL53L1X_GetDistance(dev, &distance);
+        status = VL53L1X_ClearInterrupt(dev);
+        AverageDistance = AverageDistance + distance;
+    }
+    status = VL53L1X_StopRanging(dev);
+    AverageDistance = AverageDistance / 50;
+    *offset = TargetDistInMm - AverageDistance;
+    status = VL53L1_WrWord(dev, ALGO__PART_TO_PART_RANGE_OFFSET_MM, *offset*4);
+    return status;
+}
+
+int8_t VL53L1X_CalibrateXtalk(busDevice_t * dev, uint16_t TargetDistInMm, uint16_t *xtalk)
+{
+    uint8_t i, tmp;
+    float AverageSignalRate = 0;
+    float AverageDistance = 0;
+    float AverageSpadNb = 0;
+    uint16_t distance = 0, spadNum;
+    uint16_t sr;
+    VL53L1X_ERROR status = 0;
+    uint32_t calXtalk;
+
+    status = VL53L1_WrWord(dev, 0x0016,0);
+    status = VL53L1X_StartRanging(dev);
+    for (i = 0; i < 50; i++) {
+        tmp = 0;
+        while (tmp == 0){
+            status = VL53L1X_CheckForDataReady(dev, &tmp);
+        }
+        status= VL53L1X_GetSignalRate(dev, &sr);
+        status= VL53L1X_GetDistance(dev, &distance);
+        status = VL53L1X_ClearInterrupt(dev);
+        AverageDistance = AverageDistance + distance;
+        status = VL53L1X_GetSpadNb(dev, &spadNum);
+        AverageSpadNb = AverageSpadNb + spadNum;
+        AverageSignalRate =
+            AverageSignalRate + sr;
+    }
+    status = VL53L1X_StopRanging(dev);
+    AverageDistance = AverageDistance / 50;
+    AverageSpadNb = AverageSpadNb / 50;
+    AverageSignalRate = AverageSignalRate / 50;
+    /* Calculate Xtalk value */
+    calXtalk = (uint16_t)(512*(AverageSignalRate*(1-(AverageDistance/TargetDistInMm)))/AverageSpadNb);
+    *xtalk = (uint16_t)((calXtalk*1000)>>9);
+    status = VL53L1_WrWord(dev, 0x0016, (uint16_t)calXtalk);
+    return status;
+}
+
+static void vl53l1x_Init(rangefinderDev_t * rangefinder)
+{
+    isInitialized = false;
+    VL53L1X_SensorInit(rangefinder->busDev);
```

Comment:
Why don't we check thet `VL53L1X_SensorInit` returned no error?

Reply:
This is addressed in most recent commit